File: articles/perseverance-and-passion-for-long-term-goals/c/perseverance_passion_score.c
#include <stdio.h>

double grit_score(double perseverance_effort, double durable_passion) {
    return 0.60 * perseverance_effort + 0.40 * durable_passion;
}

int main(void) {
    double perseverance_effort = 0.80;
    double durable_passion = 0.65;
    printf("Synthetic perseverance-passion grit score: %.3f\n",
           grit_score(perseverance_effort, durable_passion));
    return 0;
}

File: articles/perseverance-and-passion-for-long-term-goals/fortran/perseverance_passion_score.f90
program perseverance_passion_score
  implicit none
  real :: perseverance_effort, durable_passion, grit_score

  perseverance_effort = 0.80
  durable_passion = 0.65
  grit_score = 0.60 * perseverance_effort + 0.40 * durable_passion

  print *, "Synthetic perseverance-passion grit score:", grit_score
end program perseverance_passion_score
